### PR TITLE
[IT-2460] Generate category rules macro snippet

### PR DESCRIPTION
**Pull request by jesusaurus**
Add an API endpoint for generating a macro snippet to be consumed by cfn-macro-cost-rules.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ boto3 = "~=1.24"
 
 [packages]
 requests = "~=2.0"
+"ruamel.yaml" = "~=0.17"
```
Comment (zaro0508) [inline Pipfile:15]: should there be quotes around ruamel.yaml?

File: mips_api/__init__.py [hunk 2]
```diff
@@ -230,21 +419,31 @@ def _build_return(code, body):
...
-                # return valid tags
-                return _build_return(200, valid_tags)
+            elif event_path == api_routes['ApiCostCategoryRules']:
+                try:
+                    # scan account tags for account codes
+                    account_codes = collect_account_tag_codes(aws_tags)
+
+                    # build the rule snippet
+                    rule_snippet = list_rules(mips_chart, aws_tags, account_codes)
+                    return _build_return(200, obj=rule_snippet, ctype='yaml')
```
Comment (zaro0508) [inline mips_api/__init__.py:439]: I would return in json and let the user handle the conversion if needed.  The other option is to create a `format` parameter on the endpoint to allow the user to specify the return format (i.e. https://abc.org/rules?format=yaml)

File: mips_api/__init__.py [hunk 3]
```diff
@@ -164,11 +272,88 @@ def list_tags(params, chart_dict, omit_list, extra_dict):
...
+        "statusCode": code,
+    }
+
+    headers = {}
+    if ctype == 'json':
+        headers['content-type'] = "text/json; charset=utf-8"
+    elif ctype == 'yaml':
+        headers['content-type'] = "text/vnd.yaml; charset=utf-8"
+    else:
+        headers['content-type'] = "text/plain; charset=utf-8"
```
Comment (zaro0508) [inline mips_api/__init__.py:336]: i think it would be better to make default or fall thru json instead of text.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: mips_api/__init__.py.orig [hunk 4]
```diff
@@ -0,0 +1,407 @@
+import copy
```
[the remark this answers is not in the document]
Comment (jesusaurus) [reply mips_api/__init__.py.orig:1]: wow, whoops, I accidentally added one of the rebase conflict files during one of my rebases.

File: Pipfile (hunk 1 above)
Comment (jesusaurus) [reply Pipfile:15]: yeah, the `.` was confusing pipenv, and I needed this for pipenv to find the package.

**Comment by jesusaurus** [comment]
superseded by https://github.com/Sage-Bionetworks-IT/lambda-finops-cost-rules/pull/1

**Pull request closed without merging**